mpack-instance-manager/src/main/python/instance_manager/instance_manager.py

```diff
@@ -0,0 +1,500 @@
+"""
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+
+__all__ = ["create_mpack", "set_mpack_instance", "get_conf_dir", "list_instances"]
+
+import sys
+import os
+import json
+
+MPACK_JSON_FILE_NAME = 'mpack.json'
+CURRENT_SOFTLINK_NAME = 'current'
+CONFIGS_DIRECTORY_NAME = 'conf'
+
+ROOT_FOLDER_PATH = "/opt/odp"
+
+INSTANCES_FOLDER_NAME = "instances"
+MODULES_FOLDER_NAME = "modules"
+MPACKS_FOLDER_NAME = "mpacks"
+DEFAULT_COMPONENT_INSTANCE_NAME = 'default'
+DEFAULT_MPACK_INSTANCE_NAME = 'default'
+
+
+def create_mpack(mpack_name, mpack_version, mpack_instance, subgroup_name, module_name,
```

Comment:
subgroup_name should be an optional parameter.

Reply:
If not provided, subgroup-name would be named "default".